storage/database.go

```diff
@@ -264,20 +257,38 @@ func (d *db) IsBootstrapped() bool {
 	return d.bsm.IsBootstrapped()
 }
 
-func (d *db) getOwnedShards() []databaseShard {
+func (d *db) TruncateNamespace(namespace string) error {
+	n, err := d.readableNamespace(namespace)
+	if err != nil {
+		return err
+	}
+	n.Truncate()
+	return nil
+}
+
+func (d *db) readableNamespace(namespace string) (databaseNamespace, error) {
 	d.RLock()
-	// If the database is not open, don't return anything.
-	if d.state != databaseOpen {
+	if !d.bsm.IsBootstrapped() {
 		d.RUnlock()
-		return nil
+		return nil, errDatabaseNotBootstrapped
 	}
-	shards := d.shardSet.Shards()
-	databaseShards := make([]databaseShard, len(shards))
-	for i, shard := range shards {
-		databaseShards[i] = d.shards[shard]
+	n, exists := d.namespaces[namespace]
+	d.RUnlock()
+
+	if !exists {
+		return nil, fmt.Errorf("not responsible for namespace %s", namespace)
```

Comment:
Perhaps better namespace error would be `fmt.Errorf("no such namespace: %s", namespace)`